scripts/nbconverted/explore_relationship_AandB.py

```diff
@@ -0,0 +1,585 @@
+    """
+
+    # Collect the extreme gene expressions
+    # Get sample IDs with the lowest 5% of reference gene expression
+    threshold_low = np.percentile(target_gene_sorted, percent_low)
+    lowest = target_gene_sorted[target_gene_sorted <= threshold_low]
+
+    # Get sample IDs with the highest 5% of reference gene expression
+    threshold_high = np.percentile(target_gene_sorted, percent_high)
+    highest = target_gene_sorted[target_gene_sorted >= threshold_high]
+
+    # Average gene expression across samples in each extreme group
+    lowest_mean = (lowest.values).mean()
+    highest_mean = (highest.values).mean()
+
+    # Different in extremes
+    denom = highest_mean - lowest_mean
+
+    # scale_factor is the proportion along the gene expression gradient
+    scale_factor = expression_profile / denom
+
+    return scale_factor
+
+
+# In[10]:
+
+
+# Apply function 
+out_dir = os.path.join(base_dir, "output", analysis_name)
+encoded_dir = os.path.join(base_dir, "encoded", analysis_name)
+
+percent_low = 5
+percent_high = 95
+interpolate_in_vae_latent_space_AB(sim_data,
+                                   test_samples_sorted,
+                                   model_encoder_file,
+                                   model_decoder_file,
+                                   weights_encoder_file,
+                                   weights_decoder_file,
+                                   encoded_dir,
+                                   rep_gene_A,
+                                   percent_low,
+                                   percent_high,
+                                   out_dir)
+
+
+# ## Plot
+# Plot gene expression A vs mean expression of genes in set B after transformation
+# What is the relationship between genes in set A and B?  As the expression of A varies how does the expression of B vary?
+
+# In[11]:
+
+
+# Read dataframe with gene expression transformed
+predict_file = os.path.join(base_dir, "output", analysis_name, "predicted_gene_exp.txt")
+predict_gene_exp = pd.read_table(predict_file, header=0, index_col=0)
+
+print(predict_gene_exp.shape)
+predict_gene_exp.head()
+
+
+# In[12]:
+
+
+# Get the means of B genes
+
+# Convert dataframe with gene ids to list
+geneSetB_ls = [l[0] for l in geneSetB.values.tolist()]
+
+geneSetB_exp = predict_gene_exp[geneSetB_ls]
+
+# Get the mean for each sample
+geneSetB_mean = geneSetB_exp.mean(axis=1)
+geneSetB_mean.head()
+
+
+# In[13]:
+
+
+# Join original expression of A and mean(transformed expression of B)
+original_A_exp = test_samples[rep_gene_A]
+predict_B_mean_exp = geneSetB_mean
+
+A_and_B_df = pd.merge(original_A_exp.to_frame('gene A untransformed'),
+                      predict_B_mean_exp.to_frame('mean gene B transformed'),
+                      left_index=True, right_index=True)
+A_and_B_df.head()
+
+
+# In[14]:
+
+
+# Plot
+sns.regplot(x='gene A untransformed',
```

Comment:
Yay it is linear! is that expected? I.e. is this a good result?

Reply:
This plot shows that the relationship that is learned by the VAE appears to be linear, which is NOT the relationship that we put into the dataset (remember we put a step function relationship).  So now the question is why the VAE is learning a linear relationship between A and B genes?  Or is this an artifact of how we plotted the data?